datastore-benchmarks/results.md

```diff
@@ -0,0 +1,84 @@
+# Results
+
+The following results were obtained on a 2019 MacBook Pro with a 2.8 GHz Intel Core i7 and 16 GB of 2133 MHz RAM.
+
+| N = 5000 | Yjs | lumino | automerge |
```

Comment:
I compiled the benchmarks again with simple JSON encoding for lumino. Let me know if you want to receive a PR with that approach. But here you can see that encoding really makes a difference.. In some cases the `parseTime` is now 10x slower.

The test case `N = 5000` only creates a small amount of document changes. 5000 changes is nothing in a real-world scenario. 

However, when I increase N to as little as 30k, the lumino test-case runs out of memory at B1.3. 

| N = 5000 | Yjs | lumino | automerge |
| :- | -: | -: | -: |
|Bundle size                                                               |     66310 bytes |                 |    251368 bytes |
|Bundle size (gzipped)                                                     |     19491 bytes |                 |     59856 bytes |
|[B1.1] Append N characters (time)                                         |          160 ms |          141 ms |                 |
|[B1.1] Append N characters (avgUpdateSize)                                |        20 bytes |        44 bytes |                 |
|[B1.1] Append N characters (docSize)                                      |      5018 bytes |    220000 bytes |                 |
|[B1.1] Append N characters (parseTime)                                    |            0 ms |           47 ms |                 |
|[B1.1] Append N characters                                                |                 |                 |        skipping |
|[B1.2] Insert string of length N (time)                                   |            2 ms |           18 ms |                 |
|[B1.2] Insert string of length N (avgUpdateSize)                          |      5018 bytes |     90026 bytes |                 |
|[B1.2] Insert string of length N (docSize)                                |      5018 bytes |     90026 bytes |                 |
|[B1.2] Insert string of length N (parseTime)                              |            0 ms |            9 ms |                 |
|[B1.2] Insert string of length N                                          |                 |                 |        skipping |
|[B1.3] Prepend N characters (time)                                        |          123 ms |         1113 ms |                 |
|[B1.3] Prepend N characters (avgUpdateSize)                               |        20 bytes |      7906 bytes |                 |
|[B1.3] Prepend N characters (docSize)                                     |     49881 bytes |  39528480 bytes |                 |
|[B1.3] Prepend N characters (parseTime)                                   |            4 ms |          783 ms |                 |
|[B1.3] Prepend N characters                                               |                 |                 |        skipping |
|[B1.4] Insert N characters at random positions (time)                     |           93 ms |           84 ms |                 |
|[B1.4] Insert N characters at random positions (avgUpdateSize)            |        27 bytes |        60 bytes |                 |
|[B1.4] Insert N characters at random positions (docSize)                  |     83706 bytes |    300032 bytes |                 |
|[B1.4] Insert N characters at random positions (parseTime)                |            2 ms |           46 ms |                 |
|[B1.4] Insert N characters at random positions                            |                 |                 |        skipping |
|[B1.5] Insert N words at random positions (time)                          |           84 ms |          292 ms |                 |
|[B1.5] Insert N words at random positions (avgUpdateSize)                 |        33 bytes |       309 bytes |                 |
|[B1.5] Insert N words at random positions (docSize)                       |    168806 bytes |   1543942 bytes |                 |
|[B1.5] Insert N words at random positions (parseTime)                     |            6 ms |          153 ms |                 |
|[B1.5] Insert N words at random positions                                 |                 |                 |        skipping |
|[B1.6] Insert string, then delete it (time)                               |            1 ms |            7 ms |                 |
|[B1.6] Insert string, then delete it (avgUpdateSize)                      |      5029 bytes |    180026 bytes |                 |
|[B1.6] Insert string, then delete it (docSize)                            |        27 bytes |    180026 bytes |                 |
|[B1.6] Insert string, then delete it (parseTime)                          |            0 ms |            6 ms |                 |
|[B1.6] Insert string, then delete it                                      |                 |                 |        skipping |
|[B1.7] Insert/Delete strings at random positions (time)                   |          142 ms |          142 ms |                 |
|[B1.7] Insert/Delete strings at random positions (avgUpdateSize)          |        23 bytes |       380 bytes |                 |
|[B1.7] Insert/Delete strings at random positions (docSize)                |     78132 bytes |   1897530 bytes |                 |
|[B1.7] Insert/Delete strings at random positions (parseTime)              |            5 ms |           70 ms |                 |
|[B1.7] Insert/Delete strings at random positions                          |                 |                 |        skipping |
|[B1.8] Append N numbers (time)                                            |          113 ms |          218 ms |                 |
|[B1.8] Append N numbers (avgUpdateSize)                                   |        23 bytes |        50 bytes |                 |
|[B1.8] Append N numbers (docSize)                                         |     29701 bytes |    250000 bytes |                 |
|[B1.8] Append N numbers (parseTime)                                       |            0 ms |          107 ms |                 |
|[B1.8] Append N numbers                                                   |                 |                 |        skipping |
|[B1.9] Insert Array of N numbers (time)                                   |            1 ms |            6 ms |                 |
|[B1.9] Insert Array of N numbers (avgUpdateSize)                          |     29742 bytes |    120026 bytes |                 |
|[B1.9] Insert Array of N numbers (docSize)                                |     29742 bytes |    120026 bytes |                 |
|[B1.9] Insert Array of N numbers (parseTime)                              |            0 ms |            4 ms |                 |
|[B1.9] Insert Array of N numbers                                          |                 |                 |        skipping |
|[B1.10] Prepend N numbers (time)                                          |           50 ms |         1264 ms |                 |
|[B1.10] Prepend N numbers (avgUpdateSize)                                 |        25 bytes |      7928 bytes |                 |
|[B1.10] Prepend N numbers (docSize)                                       |     74527 bytes |  39639168 bytes |                 |
|[B1.10] Prepend N numbers (parseTime)                                     |            1 ms |          858 ms |                 |
|[B1.10] Prepend N numbers                                                 |                 |                 |        skipping |
|[B1.11] Insert N numbers at random positions (time)                       |           84 ms |          264 ms |                 |
|[B1.11] Insert N numbers at random positions (avgUpdateSize)              |        32 bytes |        68 bytes |                 |
|[B1.11] Insert N numbers at random positions (docSize)                    |    108311 bytes |    341776 bytes |                 |
|[B1.11] Insert N numbers at random positions (parseTime)                  |            2 ms |          123 ms |                 |
|[B1.11] Insert N numbers at random positions                              |                 |                 |        skipping |
|[B2.1] Concurrently insert string of length N at index 0 (time)           |            1 ms |           39 ms |                 |
|[B2.1] Concurrently insert string of length N at index 0 (updateSize)     |     10036 bytes |    180052 bytes |                 |
|[B2.1] Concurrently insert string of length N at index 0 (docSize)        |     10142 bytes |     91852 bytes |                 |
|[B2.1] Concurrently insert string of length N at index 0 (parseTime)      |            0 ms |            6 ms |                 |
|[B2.1] Concurrently insert string of length N at index 0                  |                 |                 |        skipping |
|[B2.2] Concurrently insert N characters at random positions (time)        |           60 ms |          308 ms |                 |
|[B2.2] Concurrently insert N characters at random positions (updateSize)  |    167207 bytes |    250580 bytes |                 |
|[B2.2] Concurrently insert N characters at random positions (docSize)     |    168105 bytes |    126844 bytes |                 |
|[B2.2] Concurrently insert N characters at random positions (parseTime)   |            4 ms |           18 ms |                 |
|[B2.2] Concurrently insert N characters at random positions               |                 |                 |        skipping |
|[B2.3] Concurrently insert N words at random positions (time)             |           65 ms |          928 ms |                 |
|[B2.3] Concurrently insert N words at random positions (updateSize)       |    335255 bytes |   2082436 bytes |                 |
|[B2.3] Concurrently insert N words at random positions (docSize)          |    336033 bytes |   1043276 bytes |                 |
|[B2.3] Concurrently insert N words at random positions (parseTime)        |           29 ms |          110 ms |                 |
|[B2.3] Concurrently insert N words at random positions                    |                 |                 |        skipping |
|[B2.4] Concurrently insert & delete (time)                                |          115 ms |         2174 ms |                 |
|[B2.4] Concurrently insert & delete (updateSize)                          |    514510 bytes |   5333360 bytes |                 |
|[B2.4] Concurrently insert & delete (docSize)                             |    515176 bytes |   2761910 bytes |                 |
|[B2.4] Concurrently insert & delete (parseTime)                           |           32 ms |          226 ms |                 |
|[B2.4] Concurrently insert & delete                                       |                 |                 |        skipping |
|[B3.1] √N clients concurrently set number in Map (time)                   |            5 ms |           16 ms |                 |
|[B3.1] √N clients concurrently set number in Map (updateSize)             |      1402 bytes |      3080 bytes |                 |
|[B3.1] √N clients concurrently set number in Map (docSize)                |      1668 bytes |        44 bytes |                 |
|[B3.1] √N clients concurrently set number in Map (parseTime)              |            1 ms |            0 ms |                 |
|[B3.1] √N clients concurrently set number in Map                          |                 |                 |        skipping |
|[B3.2] √N clients concurrently set Object in Map (time)                   |            6 ms |           17 ms |                 |
|[B3.2] √N clients concurrently set Object in Map (updateSize)             |      4108 bytes |      3340 bytes |                 |
|[B3.2] √N clients concurrently set Object in Map (docSize)                |      1845 bytes |        46 bytes |                 |
|[B3.2] √N clients concurrently set Object in Map (parseTime)              |            1 ms |            0 ms |                 |
|[B3.2] √N clients concurrently set Object in Map                          |                 |                 |        skipping |
|[B3.3] √N clients concurrently set String in Map (time)                   |            3 ms |           10 ms |                 |
|[B3.3] √N clients concurrently set String in Map (updateSize)             |     10556 bytes |     20720 bytes |                 |
|[B3.3] √N clients concurrently set String in Map (docSize)                |      1738 bytes |       176 bytes |                 |
|[B3.3] √N clients concurrently set String in Map (parseTime)              |            0 ms |            0 ms |                 |
|[B3.3] √N clients concurrently set String in Map                          |                 |                 |        skipping |
|[B3.4] √N clients concurrently insert text in Array (time)                |            5 ms |           12 ms |                 |
|[B3.4] √N clients concurrently insert text in Array (updateSize)          |      1525 bytes |      3200 bytes |                 |
|[B3.4] √N clients concurrently insert text in Array (docSize)             |      1387 bytes |        44 bytes |                 |
|[B3.4] √N clients concurrently insert text in Array (parseTime)           |            0 ms |            0 ms |                 |
|[B3.4] √N clients concurrently insert text in Array                       |                 |                 |        skipping |